### PR TITLE
use unique indexes to allow deletion of pages

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: scripts/create-tables.sql [hunk 1]
```diff
@@ -56,6 +55,13 @@ CREATE TABLE IF NOT EXISTS "elements"
   CONSTRAINT elements_id_pk PRIMARY KEY (id)
 );
 
+/* Unique indexes */
+CREATE UNIQUE INDEX pages_project_id_x_y_key ON pages
+(project_id, x, y) WHERE deleted_at IS NULL;
+
+CREATE UNIQUE INDEX pages_project_id_x_y_deleted_at_key ON pages
+(project_id, x, y, deleted_at) WHERE deleted_at IS NOT NULL;
```
Comment (thisandagain) [inline scripts/create-tables.sql]: You don't want this because subsequent `DELETES` for the same page will fail.

**Comment by thisandagain** [comment]
I proposed a similar but slightly different fix in #46 . Let me know what you think.

File: scripts/create-tables.sql (hunk 1 above)
Comment (jbuck) [inline scripts/create-tables.sql]: Why would they fail? Wouldn't the deleted_at value be different for each deleted versions?

Comment (k88hudson) [inline scripts/create-tables.sql]: Yeah I don't think it's going to fail, but it's not really necessary either

**Pull request closed without merging**